glue/qt/widgets/custom_component_widget.py

```diff
@@ -114,12 +201,25 @@ def create_component(collection):
         """
         # pylint: disable=W0212
         widget = CustomComponentWidget(collection)
-        widget.ui.show()
-        if widget.ui.exec_() == QDialog.Accepted:
-            link = widget._create_link()
-            if link:
-                widget._add_link_to_targets(link)
-
+        while True:
+            widget.ui.show()
+            if widget.ui.exec_() == QDialog.Accepted:
+                if len(str(widget.ui.expression.toPlainText())) == 0:
+                    QMessageBox.critical(widget.ui, "Error", "No expression set",
+                                         buttons=QMessageBox.Ok)                    
+                elif widget._number_targets == 0:
+                    QMessageBox.critical(widget.ui, "Error", "Please specify the target dataset(s)",
+                                         buttons=QMessageBox.Ok)
+                elif len(widget.ui.new_label.text()) == 0:
+                    QMessageBox.critical(widget.ui, "Error", "Please specify the new component name",
+                                         buttons=QMessageBox.Ok)
```

Comment:
I think this can be combined with https://github.com/glue-viz/glue/issues/577 which will require calculating values before the dialog is closed anyway.